File: frontend/src/types/branch.ts
export interface Branch {
  id: number;
  branch_code: string;
  branch_name: string;
  address: string;
  city: string;
  latitude: number;
  longitude: number;
  phone: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface BranchCreate {
  branch_code: string;
  branch_name: string;
  address: string;
  city: string;
  latitude: number;
  longitude: number;
  phone?: string | null;
  status: string;
}

export interface BranchUpdate {
  branch_code?: string;
  branch_name?: string;
  address?: string;
  city?: string;
  latitude?: number;
  longitude?: number;
  phone?: string | null;
  status?: string;
}

File: backend/app/models/branch.py
from datetime import datetime, timezone

from sqlalchemy import DateTime, Float, String
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class Branch(Base):
    __tablename__ = "branches"

    id: Mapped[int] = mapped_column(
        primary_key=True,
        index=True
    )

    branch_code: Mapped[str] = mapped_column(
        String(20),
        unique=True,
        nullable=False,
        index=True
    )

    branch_name: Mapped[str] = mapped_column(
        String(100),
        nullable=False
    )

    address: Mapped[str] = mapped_column(
        String(255),
        nullable=False
    )

    city: Mapped[str] = mapped_column(
        String(100),
        nullable=False
    )

    latitude: Mapped[float] = mapped_column(
        Float,
        nullable=False
    )

    longitude: Mapped[float] = mapped_column(
        Float,
        nullable=False
    )

    phone: Mapped[str | None] = mapped_column(
        String(20),
        nullable=True
    )

    status: Mapped[str] = mapped_column(
        String(20),
        default="ACTIVE",
        nullable=False
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=lambda: datetime.now(timezone.utc),
        nullable=False
    )

    updated_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=lambda: datetime.now(timezone.utc),
        onupdate=lambda: datetime.now(timezone.utc),
        nullable=False
    )